extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -212,41 +213,191 @@ void discoverBindableServices(BuildProducer<BindableServiceBuildItem> bindables,
                 continue;
             }
             BindableServiceBuildItem item = new BindableServiceBuildItem(service.name());
-            Set<MethodInfo> blockingMethods = gatherBlockingMethods(service);
-            for (MethodInfo method : blockingMethods) {
-                item.registerBlockingMethod(method.name());
+            Set<String> blockingMethods = gatherBlockingMethods(service, index);
+            for (String method : blockingMethods) {
+                item.registerBlockingMethod(method);
             }
             bindables.produce(item);
         }
     }
 
-    private Set<MethodInfo> gatherBlockingMethods(ClassInfo service) {
-        Set<MethodInfo> result = new HashSet<>();
-        boolean blockingClass = service.classAnnotation(GrpcDotNames.BLOCKING) != null;
-        if (blockingClass && service.classAnnotation(GrpcDotNames.NON_BLOCKING) != null) {
-            throw new DeploymentException("Class '" + service.name()
-                    + "' contains both @Blocking and @NonBlocking annotations.");
+    /**
+     * Generate list of {@link ClassInfo} with {@code service} as the first element and the class implementing
+     * {@code io.grpc.BindableService} (the protobuf generated {@code *ImplBase}) as the last one.
```

Comment:
Note that [user beans that implement service interface](https://quarkus.io/guides/grpc-service-implementation#implementing-a-service-with-the-mutiny-api) do not have to implement `BindableService` at all.

Reply:
Updated the javadoc + inline comment